ADD: 플레이어 시작 위치 추가, 포탈 생성 위치 수정

Jo-sm94 opened the pull request:
BP_PlayerSpawn
 - 플레이어 시작 위치와 방향을 정해주는 액터

GameMode
 - 포탈 생성 위치 기본값 0, 0에서 포탈 생성 시 플레이어의 위치가 가까울 경우 생성 위치 변경 로직

Bathboomsoso committed:
Merge pull request #36 from NbcampUnreal/feature/Animfixed

Feature/animfixed

Jo-sm94 committed:
Merge pull request #38 from NbcampUnreal/Develop

merge

Jo-sm94 committed:
Merge remote-tracking branch 'origin/Develop' into feature/GameMode01

Jo-sm94 committed:
Merge remote-tracking branch 'origin/Develop' into feature/GameMode01

Jo-sm94 committed:
ADD: 플레이어 시작 위치 추가, 포탈 생성 위치 수정

BP_PlayerSpawn
 - 플레이어 시작 위치와 방향을 정해주는 액터

GameMode
 - 포탈 생성 위치 기본값 0, 0에서 포탈 생성 시 플레이어의 위치가 가까울 경우 생성 위치 변경 로직

hwarang98 commented on Source/Shooter/Private/GameModes/ShooterBaseGameMode.cpp:
```diff
@@ -159,15 +158,46 @@ void AShooterBaseGameMode::OnAllEnemiesDefeated()
+		return DefaultPortalLocation;
+	}
+
+	ACharacter* PlayerCharacter = UGameplayStatics::GetPlayerCharacter(World ,0);
```
여기 nullcheck 필요할것같습니다!

hwarang98 commented on Source/Shooter/Private/GameModes/ShooterBaseGameMode.cpp:
```diff
@@ -159,15 +158,46 @@ void AShooterBaseGameMode::OnAllEnemiesDefeated()
+
+	FVector PortalLocation;
+
+	if (Dist < 250.0f)
```
250도 하드코딩보단 언리얼에디터에서 수정가능한값으로 수정하면 좋을것같습니다! 아래의 `PortalLocation = PlayerLoc + Dir * 400.0f;` 에서 400도 마찬가지입니다! 혹시 다른 의도가있다면 그대로 두시져!

Jo-sm94 committed:
에디터 변수 설정

BP_PlayerSpawn
 - 플레이어 시작 위치와 방향을 정해주는 액터

GameMode
 - 포탈 생성 위치 기본값 0, 0에서 포탈 생성 시 플레이어의 위치가 가까울 경우 생성 위치 변경 로직
 - 에디터 변수 추가 : 캐릭터와 포탈 생성 위치의 거리 차이 기준 수치, 포탈 생성 위치 변경 수치
